### ISSUE TITLE
Use datapacks for recipes, spells, etc.

### ISSUE THREAD

**Issue by slava110** [Team-Eidolon/eidolon#24]
Hi. I'd like to add custom recipes to dark touch but it's currently impossible. And a lot of stuff isn't accessible for modpack developers at all. I think that datapack support would open **a lot** of possibilities for modpack developers c:
These things could probably be in datapacks:
- [ ] Custom researches
- [ ] Custom altar empowering stuff
- [ ] Touch of darkness recipes
- [ ] Custom mystical signs
- [ ] Custom runes
- [ ] Custom chants
- [ ] Custom rituals

Having KubeJS support would be nice too

**Comment by QinomeD** [comment]
Sounds cool, but the mod itself is barely usable, you can't even chant yet in survival without cheats

**Comment by vertexcubed** [comment]
Would probably be a good idea to migrate runes and research to be data driven once research is a bit more usable. 

Rituals are a little harder due to their effects being more complicated than simply crafting (though I think occultism does custom rituals despite this, so could be possible). However, you can still make custom rituals with KubeJS using reflection (though they'd probably have to be registered at startup), if you want help with that just message me on discord.

Custom mystical signs might be kind of hard to make data driven, same with chants. Touch of Darkness doesn't work yet afaik, but that could be made data driven as well probably.

**Comment by slava110** [comment]
> However, you can still make custom rituals with KubeJS using reflection

Yeah, yesterday I've tried to use KubeJS for everything mentioned and it worked. Would be cool to be able to do it without reflection though.

> Touch of Darkness doesn't work yet afaik

It's working. There's just no way to get runes in survival and illwood is useless (illwood logs don't even drop)
